src/main/java/seedu/address/model/event/EventDate.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+public class EventDate {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be valid and in format YYYY-MM-DD not before today's date.";
+
+    public static final String VALIDATION_REGEX = "\\d{4}-\\d{2}-\\d{2}";
+
+    public final String eventDate;
+
+    /**
+     * Constructs a {@code EventDate}.
+     *
+     * @param date A valid EventDate.
+     */
+    public EventDate(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
+        eventDate = date;
+    }
+
+    /**
+     * Returns true if a given string is a valid date.
+     */
+    public static boolean isValidDate(String test) {
```

Comment:
Can have a better naming for the argument.
```suggestion
    public static boolean isValidDate(String testDate) {
```

Reply:
fixed